core/src/banking_stage/consumer.rs

```diff
@@ -1074,7 +1082,9 @@ mod tests {
             mint_keypair,
             ..
         } = create_slow_genesis_config(10_000);
-        let bank = Arc::new(Bank::new_no_wallclock_throttle_for_tests(&genesis_config));
+        let mut bank = Bank::new_no_wallclock_throttle_for_tests(&genesis_config);
+        bank.deactivate_feature(&feature_set::apply_cost_tracker_during_replay::id());
```

Comment:
Any reason to explicitly **disable** the feature for this test? I don't see anything in it that stands out as failing immediately.
If that's the case, imo it's better to have this test run the scenario twice with the feature enabled and disabled.

But obviously let me know if I'm missing something!

Reply:
per the intention of the pr, with the feature enabled, the adjustment will be bypassed, which breaks this test.

But a good point to test it with out both feature on/off. Will PR with added test.

Reply:
But what part of the test breaks without this? All the asserts check, afaict, is that there is *some* non-zero cost, and the tx was committed.

Reply:
It breaks here:
```
-            assert_eq!(get_block_cost(), 2 * single_transfer_cost);
-            assert_eq!(get_tx_count(), 2);
+            assert_eq!(get_block_cost(), expected_block_cost);
+            assert_eq!(get_tx_count(), expected_tracked_tx_count);
```
the adjustment logic removes failed tx (the one fail due to AccountInUse) from cost tracker. #31708 for it.